src/CMakeLists.txt

```diff
@@ -0,0 +1,77 @@
+# minimum required CMAKE version
+CMAKE_MINIMUM_REQUIRED(VERSION 3.7 FATAL_ERROR)
+
+project(parse_fasm)
+
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+
+set(CMAKE_CXX_STANDARD 11)
+
+# add antrl4cpp artifacts to project environment
+include_directories(${ANTLR4_INCLUDE_DIRS})
+
+# set variable pointing to the antlr tool that supports C++
+# this is not required if the jar file can be found under PATH environment
+set(ANTLR_EXECUTABLE ${CMAKE_CURRENT_SOURCE_DIR}/../third_party/antlr/antlr-4.9-complete.jar)
+
+# add macros to generate ANTLR Cpp code from grammar
+find_package(ANTLR REQUIRED)
+
+find_library(ANTLR4_RUNTIME NAMES antlr4 antlr4-runtime REQUIRED)
+
+# Unit testing library
+find_package(Catch2 REQUIRED)
+
+# Lexer and parser targets
+antlr_target(FasmLexer antlr/FasmLexer.g4 LEXER
+  OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+  COMPILE_FLAGS -Xexact-output-dir
+  )
+antlr_target(FasmParser antlr/FasmParser.g4 PARSER VISITOR
+  OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+  DEPENDS_ANTLR FasmLexer
+  COMPILE_FLAGS -Xexact-output-dir -lib ${ANTLR_FasmLexer_OUTPUT_DIR}
+  )
+
+# Explanation:
+# line 1: Skip lines starting in #define
+#      2: Extract TAGS(...) from dependencies
+#      3: Convert from TAGS('c', long_name) -> long_name = b'c, write to file
+add_custom_target(tags.py ALL
+  COMMAND grep -ve ^\#define ${CMAKE_CURRENT_SOURCE_DIR}/ParseFasm.cpp |
+          grep -hoe TAG\([^\)]*\) |
+          sed -e s/TAG\(\\\(.*\\\),\ \\\(.*\\\)\)/\\2\ =\ b\\1/ > tags.py
+  DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/ParseFasm.cpp
+  VERBATIM
+  )
+
+# Include generated files in project environment
+include_directories(${ANTLR_FasmLexer_OUTPUT_DIR})
+include_directories(${ANTLR_FasmParser_OUTPUT_DIR})
+
+# Add generated grammar to binary target
+add_executable(parse_fasm_bin ParseFasm.cpp
```

Comment:
I see you repeating the same arguments to `add_executable` (2x) and `add_library` here.  Should you create a static library, and then link it to those three targets?

Reply:
I could, but there would still be a shared library, and two static copies (the test version changes the source.)
I could use the share library for the standalone version, but it's not worth the effort, since I only included the standalone parser because it might be useful for debugging, or understanding how it works. It's only 195KB.

Reply:
I've removed the standalone executable.